From 4e1ce7f7f0eabbb8c21c675c3619bc72614ac971 Mon Sep 17 00:00:00 2001
From: Thang Tran <thuutran@amperecomputing.com>
Date: Fri, 14 Jul 2023 14:58:17 +0700
Subject: [PATCH 1/4] pldm: add PldmMessagePollEvent and NumericSensorEvent

Add Numeric Sensor Event signals to inform that the event has been recevied.
Please refer Chappter 16.10 "eventData 1715 format for sensorEvent" of "PLDM for
Platform Monitoring and Control Specification".

Signed-off-by: Thang Tran <thuutran@amperecomputing.com>
---
 .../openbmc_project/PLDM/Event.interface.yaml | 76 +++++++++++++++++++
 1 file changed, 76 insertions(+)

diff --git a/yaml/xyz/openbmc_project/PLDM/Event.interface.yaml b/yaml/xyz/openbmc_project/PLDM/Event.interface.yaml
index 9206782..07f0c37 100644
--- a/yaml/xyz/openbmc_project/PLDM/Event.interface.yaml
+++ b/yaml/xyz/openbmc_project/PLDM/Event.interface.yaml
@@ -49,3 +49,79 @@ signals:
             description: >
                 The event state value for the state from which the present event
                 state was entered.
+    - name: PldmMessagePollEvent
+      description: >
+          Signal indicating that a pldmMessagePollEvent change EventMessage is
+          received.
+          More information about properties can be found at DSP0248 version
+          1.2.0 table 19.
+      properties:
+        - name: TID
+          type: byte
+          description: >
+            A terminus id.
+
+        - name: EventClass
+          type: byte
+          description: >
+            The event class value from the state change that triggered
+            the event message.
+
+        - name: FormatVersion
+          type: byte
+          description: >
+            Version of the event format
+
+        - name: EventID
+          type: uint16
+          description: >
+            Identifier for the event that requires multipart transfer
+
+        - name: DataTransferHandle
+          type: uint32
+          description: >
+            A handle that is used to identify the event data to be received
+            via the PollForPlatformEventMessage command.
+    - name: NumericSensorEvent
+      description: >
+          Signal indicating that a numeric sensor change EventMessage is
+          received.
+          More information about properties can be found at DSP0248 version
+          1.2.0 table 19.
+      properties:
+          - name: TID
+            type: byte
+            description: >
+                A terminus id.
+
+          - name: sensorID
+            type: uint16
+            description: >
+                The sensorID is the value that is used in PDRs and PLDM sensor
+                access commands to identify and access a particular sensor
+                within a terminus.
+
+          - name: eventState
+            type: byte
+            description: >
+                The event state value from the state change that triggered
+                the event message.
+
+          - name: previousEventState
+            type: byte
+            description: >
+                The event state value for the state from which the present
+                event state was entered.
+
+          - name: sensorDataSize
+            type: byte
+            description: >
+                The bit width and format of reading and threshold values that
+                the sensor returns
+
+          - name: presentReading
+            type: uint32
+            description: >
+                The present value indicated by the sensor. The sensorDataSize
+                field returns an enumeration that indicates the number of bits
+                used to return the value.
-- 
2.34.1

